activerecord/lib/active_record/relation/predicate_builder/range_handler.rb

```diff
@@ -16,15 +16,18 @@ def initialize(predicate_builder)
       def call(attribute, value)
         begin_bind = predicate_builder.build_bind_attribute(attribute.name, value.begin)
         end_bind = predicate_builder.build_bind_attribute(attribute.name, value.end)
-        if value.begin.respond_to?(:infinite?) && value.begin.infinite?
-          if value.end.respond_to?(:infinite?) && value.end.infinite?
+
+        if impossible?(begin_bind) && value.begin > 0 || impossible?(end_bind) && value.end < 0
+          attribute.in([])
+        elsif infinity?(value.begin) || !boundable?(begin_bind)
```

Comment:
Thanks for the feedback. I've added `boundable?` and `impossible?` helper methods for the range handler to make more explicit in the code.